main.go

```diff
@@ -29,194 +29,215 @@ import (
 
 	slogmulti "github.com/samber/slog-multi"
 	slogsyslog "github.com/samber/slog-syslog"
-	"github.com/spf13/cast"
 
+	"github.com/bank-vaults/secret-init/pkg/args"
+	"github.com/bank-vaults/secret-init/pkg/config"
+	"github.com/bank-vaults/secret-init/pkg/envstore"
 	"github.com/bank-vaults/secret-init/provider"
 	"github.com/bank-vaults/secret-init/provider/file"
+	"github.com/bank-vaults/secret-init/provider/vault"
 )
 
-func NewProvider(providerName string) (provider.Provider, error) {
+func NewProvider(providerName string, daemonMode bool) (provider.Provider, error) {
```

Comment:
you can remove the `deamonMode` argument here, I have noted in another comment on how you can get it (via `internal.LoadConfig` in places where it is needed